src/plasma/malloc.cc

```diff
@@ -129,41 +123,39 @@ int fake_munmap(void *addr, size_t size) {
   addr = pointer_retreat(addr, sizeof(size_t));
   size += sizeof(size_t);
 
-  struct mmap_record *record;
+  auto entry = mmap_records.find(addr);
 
-  HASH_FIND(hh_pointer, records_by_pointer, &addr, sizeof(addr), record);
-  if (record == NULL || record->size != size) {
+  if (entry == mmap_records.end() || entry->second.size != size) {
     /* Reject requests to munmap that don't directly match previous
      * calls to mmap, to prevent dlmalloc from trimming. */
     return -1;
   }
 
-  HASH_DELETE(hh_fd, records_by_fd, record);
-  HASH_DELETE(hh_pointer, records_by_pointer, record);
-
   int r = munmap(addr, size);
   if (r == 0) {
-    close(record->fd);
+    close(entry->second.fd);
   }
+
+  mmap_records.erase(entry);
   return r;
 }
 
 void get_malloc_mapinfo(void *addr,
                         int *fd,
                         int64_t *map_size,
                         ptrdiff_t *offset) {
-  struct mmap_record *record;
   /* TODO(rshin): Implement a more efficient search through records_by_fd. */
-  for (record = records_by_fd; record != NULL; record = record->hh_fd.next) {
-    if (addr >= record->pointer &&
-        addr < pointer_advance(record->pointer, record->size)) {
-      *fd = record->fd;
-      *map_size = record->size;
-      *offset = pointer_distance(record->pointer, addr);
+  for (const auto &entry : mmap_records) {
```

Comment:
The two hashtables were not really neccessary I think, see the TODO from Richard. I'll check back with him.